### ISSUE TITLE
1031 is not a valid program exit code for posix/systemd

### ISSUE THREAD

**Issue by arnetheduck** [status-im/nim-beacon-chain#3973]
> The value of status may be 0, EXIT_SUCCESS, EXIT_FAILURE, [[CX](javascript:open_code('CX'))] [Option Start]  or any other value, though only the least significant 8 bits (that is, status & 0377) shall be available to a waiting parent process. [Option End]

https://pubs.opengroup.org/onlinepubs/009695399/functions/exit.html

systemd in particular does not support parsing values >255:
```
Aug 16 16:37:42 raspi3 systemd[1]: /lib/systemd/system/nimbus_validator_client.service:49: Failed to parse value, ignoring: 1031
```

22.7.0

**Comment by tersec** [comment]
And it's consistent with https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Process%20Exit%20Codes too which is why per https://github.com/status-im/nimbus-eth2/issues/3687 a >1000 code was chosen to begin with, so the usable gap is a possibly non-strict subset of 8 through 199 inclusive.

**Comment by zah** [comment]
Fixed by https://github.com/status-im/nimbus-eth2/pull/3977 and shipped in Nimbus v22.8.0
